apostrophecms/apostrophe
# 3.0 notifications

# Notifications

## Issues
https://apostrophecms.atlassian.net/jira/software/projects/A30U/boards/4?assignee=5ddbce7e47baa50cfd05eed2&selectedIssue=A30U-166

https://apostrophecms.atlassian.net/jira/software/projects/A30U/boards/4?assignee=5ddbce7e47baa50cfd05eed2&selectedIssue=A30U-165

## How to test

Personnally, I added this:

```js
mounted () {
    apos.notify('New notif message', { dismiss: true });
  },
```
to `modules/@apostrophecms/piece-type/ui/apos/components/AposPiecesManagerToolbar.vue` in the `npm link`ed Apostrophe. This way, every time a modal is opened (through the admin bar, click on "Images" for example), a notification is sent from the frontend.

Otherwise, you could add this at the project level (for example in `modules/product/index.js`):

```js
  handlers(self, options) {
    return {
      afterSave: {
        test(req, doc) {
          self.apos.notify(req, 'New product added: ' + doc.title);
        }
      }
    }
  }
```
That would produce a notification from the backend.

The generated notifications should show up, and be dismissed with a click on the cross (or dismissed directly if `dismiss: true` or `dismiss` with an integer value is passed).

## Notes

@boutell We talked about `modifiedSince` but I have to admit I did not know where/when to put it as it works without it and because `PUT` and `PATCH` are voluntarily not implemented (last PR on notifications, backend side).

I also chose to `DELETE` every dismissed notification as I believe you said past notifications should be deleted (and this is why `PUT` and `PATCH` are voluntarily not implemented although `DELETE` is). If I am mistaken, we should talk about how to handle updates/dismissal.
